### PR TITLE
README: Expand readme with more details

### PR DESCRIPTION
**Pull request by shefty**
Signed-off-by: Sean Hefty <sean.hefty@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
Addresses #782  - yea, more could probably be added, but at least this is a reasonable start.  Not quite the novel that is the openmpi readme though.  :)

**Comment by jsquyres** [comment]
The OMPI README is specifically designed to cure insomnia.  We take great pride in its medicinal value.

File: README [hunk 1]
```diff
@@ -1,13 +1,78 @@
-This README is for userspace RDMA fabric library.
-
 Version @VERSION@
 Released on @DATE@
 
+Introduction
+============
+The Open Fabrics Interfaces (OFI) is a framework focused on exporting
```
Comment (jsquyres) [inline README]: This phrasing has always seemed odd to me -- "The Open Fabrics Interfaces" sounds like it is plural.  But then it says "is..." which is a singular verb.

Can we think of better phrasing?

File: README [hunk 2]
```diff
@@ -1,13 +1,78 @@
...
+OFI is being developed by the OFI Working Group (OFIWG) a subgroup
+of the OpenFabrics Alliance (OFA).  Participation in OFIWG
+(pronounced o-fee-wig) is open to anyone, regardless of their
+membership in OFA.
+
+The goal of OFI and libfabric is to define interfaces that enable
+a tight semantic map between applications and underlying fabric
+services.  Specifically, libfabric software interfaces have been
+co-designed with fabric hardware providers and application developers,
+with a focus on the needs of HPC users.  OFI supports multiple
```
Comment (jsquyres) [inline README]: How about:

...with an *initial* focus on the needs of HPC...

File: README [hunk 3]
```diff
@@ -1,13 +1,78 @@
...
+interface semantics, is fabric and hardware implementation agnostic,
+and leverages and expands the existing RDMA open source community. 
+
+For more information regarding the OFI project, please visit the OFIWG
+github site:
```
Comment (jsquyres) [inline README]: GitHub

File: README [hunk 4]
```diff
@@ -1,13 +1,78 @@
...
+OFI targets support for the Linux operating system. A reasonable effort
+is made to support all major, modern Linux distributions; however,
+validation is limited to the most recent 2-3 releases of RedHat
+Enterprise Linux (RHEL)and SUSE Linux Enterprise Server (SLES).
+OFI aligns its supported distributions with the most current
+OpenFabrics Enterprise Distribution (OFED) software releases. With
+the exception of the sockets provider, which is provided for development
+purposes, distro support for a specific provider is vendor specific.
+
+OFI will also run on OS-X, but OS-X support is provided as a convenience
```
Comment (jsquyres) [inline README]: OS X (no dash).

Comment (jsquyres) [inline README]: Also, I think you mean to say "Libfabric will also run on..."   OFI is the framework; libfabric is the implementation.

**Comment by shefty** [comment]
All issues addressed.

**Pull request merged**